### ISSUE TITLE
Fails at SBOM creation

### ISSUE THREAD

**Issue by davidkarlsen** [paketo-buildpacks/syft#189]
The executable-jar fails at a sbom step.

## Expected Behavior
Should pass as normal

## Current Behavior
```shell
2023-08-22T09:17:04.0290316Z [INFO]     [creator]     Paketo Buildpack for Executable JAR 6.7.4
2023-08-22T09:17:04.0291401Z [INFO]     [creator]       https://github.com/paketo-buildpacks/executable-jar
2023-08-22T09:17:06.5132069Z [INFO]     [creator]       2023/08/22 09:17:06 error during command execution: 1 error occurred:
2023-08-22T09:17:06.5168291Z [INFO]     [creator]       	* reflect: call of reflect.Value.Type on zero Value at:
2023-08-22T09:17:06.5168606Z [INFO]     [creator]       goroutine 20 [running]:
2023-08-22T09:17:06.5168771Z [INFO]     [creator]       runtime/debug.Stack()
2023-08-22T09:17:06.5168958Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/debug/stack.go:24 +0x65
2023-08-22T09:17:06.5274305Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.runCataloger.func1()
2023-08-22T09:17:06.5274600Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:57 +0x45
2023-08-22T09:17:06.5275537Z [INFO]     [creator]       panic({0x140bfc0, 0xc000011938})
2023-08-22T09:17:06.5276028Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/panic.go:884 +0x213
2023-08-22T09:17:06.5276288Z [INFO]     [creator]       reflect.Value.typeSlow({0x0?, 0x0?, 0x15bc1d4?})
2023-08-22T09:17:06.5276634Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/reflect/value.go:2610 +0x12e
2023-08-22T09:17:06.5276839Z [INFO]     [creator]       reflect.Value.Type(...)
2023-08-22T09:17:06.5277435Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/reflect/value.go:2605
2023-08-22T09:17:06.5277729Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.resolveProperty.func1({0xc0007cf5f0, 0x12})
2023-08-22T09:17:06.5278188Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:181 +0x4a5
2023-08-22T09:17:06.5278508Z [INFO]     [creator]       regexp.(*Regexp).ReplaceAllStringFunc.func1({0x0, 0x0, 0x0}, {0xc0017423f0?, 0xc000292c18?, 0x0?})
2023-08-22T09:17:06.5278772Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:604 +0x91
2023-08-22T09:17:06.5279030Z [INFO]     [creator]       regexp.(*Regexp).replaceAll(0xc00013ff40, {0x0, 0x0, 0x0}, {0xc0007cf5f0, 0x12}, 0x2, 0xc000292cf8)
2023-08-22T09:17:06.5279430Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:642 +0x405
2023-08-22T09:17:06.5279690Z [INFO]     [creator]       regexp.(*Regexp).ReplaceAllStringFunc(0x0?, {0xc0007cf5f0?, 0x40df27?}, 0x68?)
2023-08-22T09:17:06.5279932Z [INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:603 +0x5d
2023-08-22T09:17:06.5280256Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.resolveProperty({0x0, 0xc0002b0bd0, 0xc0006d70c0, 0x0, 0xc0002b0c10, 0x0, 0x0, 0x0, 0xc0002b0c20, 0x0, ...}, ...)
2023-08-22T09:17:06.5280584Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:152 +0x59
2023-08-22T09:17:06.5281023Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.newPackageFromPom({0x0, 0xc0002b0bd0, 0xc0006d70c0, 0x0, 0xc0002b0c10, 0x0, 0x0, 0x0, 0xc0002b0c20, 0x0, ...}, ...)
2023-08-22T09:17:06.5281345Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:83 +0x298
2023-08-22T09:17:06.5281652Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.parserPomXML({0xc0002c5a70?, 0xc0002c57c8?}, 0x415f50?, {{{{{...}, {...}}, {0x0, 0x0}, {0xea, {...}}}, {0xc000803dd0}}, ...})
2023-08-22T09:17:06.5281965Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:33 +0x351
2023-08-22T09:17:06.5282253Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/generic.(*Cataloger).Catalog(0xc000462cc0, {0x1b4d5d0, 0xc0002f6000})
2023-08-22T09:17:06.5282522Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/generic/cataloger.go:129 +0x74e
2023-08-22T09:17:06.5282801Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.runCataloger({0x1b412d8, 0xc000462cc0}, {0x1b4d5d0?, 0xc0002f6000})
2023-08-22T09:17:06.5283382Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:65 +0x1fa
2023-08-22T09:17:06.5283736Z [INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.Catalog.func1()
2023-08-22T09:17:06.5283974Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:146 +0xec
2023-08-22T09:17:06.5284193Z [INFO]     [creator]       created by github.com/anchore/syft/syft/pkg/cataloger.Catalog
2023-08-22T09:17:06.5284555Z [INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:141 +0x34a
2023-08-22T09:17:06.5284839Z [INFO]     [creator]       
2023-08-22T09:17:06.5284953Z [INFO]     [creator]       
2023-08-22T09:17:06.5285082Z [INFO]     [creator]     
2023-08-22T09:17:06.5285226Z [INFO]     [creator]     Paketo Buildpack for Executable JAR 6.7.4
2023-08-22T09:17:06.5285401Z [INFO]     [creator]       unable to create Build SBoM 
2023-08-22T09:17:06.5286197Z [INFO]     [creator]       unable to run `syft [packages -q -o json=/layers/paketo-buildpacks_executable-jar/launch.sbom.syft.json -o cyclonedx-json=/layers/paketo-buildpacks_executable-jar/launch.sbom.cdx.json dir:/workspace]`
2023-08-22T09:17:06.5286574Z [INFO]     [creator]       exit status 1
2023-08-22T09:17:06.5286926Z [INFO]     [creator]     Timer: Builder ran for 6.763050915s and ended at 2023-08-22T09:17:06Z
2023-08-22T09:17:06.5287152Z [INFO]     [creator]     ERROR: failed to build: exit status 1
2023-08-22T09:17:06.9250126Z The process '/home/vsts/work/1/s/maven/apache-maven-3.9.4/bin/mvn' failed with exit code 1
```

## Steps to Reproduce
Do a spring-boot vanilla build

## Motivations
Build fails

**Comment by dmikusa** [comment]
Thanks for the report. This looks like a bug (or at least a crash) in Syft itself. It's trying to read your pom.xml file and failing.

I just cut a new release of the syft buildpack, which pulls in a new upstream syft patch release. This might help. The other option would be trying a slightly older version of syft to see if this is a new bug that was recently introduced.

You can swap out the version by using the `pack build -b` flag. When you do a normal build, it'll print the list of participating buildpacks at the top in the DETECTION section. Take that list, and just enumerate them with the `-b` flag, so `pack build -b paketo-buildpacks/ca-certificates -b paketo-buildpacks/bellsoft-liberica ...` (your exact list will vary depending on your app).

The Syft buildpack will be in there and for it use `docker.io/paketobuildpacks/syft:latest` and that should go fetch the latest one. You can also swap `latest` with a version number from the release page. You can confirm it works because the buildpack output will print a different version number.

Please let me know what you find and we can figure out some next steps.

**Comment by davidkarlsen** [comment]
Thanks for getting back to me - I can confirm that downgrading to 1.33.0 worked - and also now the `latest` buildpack works.

**Comment by dmikusa** [comment]
Great & thanks for confirming. We'll roll out the latest buildpack with the fixed syft version in this week's release cycle, so it should be in the builders on Friday.

**Comment by davidkarlsen** [comment]
Sorry - I was a bit quick, it still fails:
```shell
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Syft 1.34.0
[INFO]     [creator]       https://github.com/paketo-buildpacks/syft
[INFO]     [creator]         Downloading from https://github.com/anchore/syft/releases/download/v0.87.0/syft_0.87.0_linux_amd64.tar.gz
[INFO]     [creator]         Verifying checksum
[INFO]     [creator]         Writing env.build/SYFT_CHECK_FOR_APP_UPDATE.default
[INFO]     [creator]     
[INFO]     [creator]     Paketo Buildpack for Executable JAR 6.7.4
[INFO]     [creator]       https://github.com/paketo-buildpacks/executable-jar
[INFO]     [creator]       2023/08/24 07:42:10 error during command execution: 1 error occurred:
[INFO]     [creator]       	* reflect: call of reflect.Value.Type on zero Value at:
[INFO]     [creator]       goroutine 22 [running]:
[INFO]     [creator]       runtime/debug.Stack()
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/debug/stack.go:24 +0x65
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.runCataloger.func1()
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:57 +0x45
[INFO]     [creator]       panic({0x140bfc0, 0xc0004ff0c8})
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/panic.go:884 +0x213
[INFO]     [creator]       reflect.Value.typeSlow({0x0?, 0x0?, 0x15bc1d4?})
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/reflect/value.go:2610 +0x12e
[INFO]     [creator]       reflect.Value.Type(...)
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/reflect/value.go:2605
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.resolveProperty.func1({0xc0008595a8, 0x12})
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:181 +0x4a5
[INFO]     [creator]       regexp.(*Regexp).ReplaceAllStringFunc.func1({0x0, 0x0, 0x0}, {0xc000876f90?, 0xc0004eec18?, 0x0?})
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:604 +0x91
[INFO]     [creator]       regexp.(*Regexp).replaceAll(0xc0000ed360, {0x0, 0x0, 0x0}, {0xc0008595a8, 0x12}, 0x2, 0xc0004eecf8)
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:642 +0x405
[INFO]     [creator]       regexp.(*Regexp).ReplaceAllStringFunc(0x40b54b?, {0xc0008595a8?, 0xc000101310?}, 0xc0004eeec8?)
[INFO]     [creator]       	/opt/hostedtoolcache/go/1.20.7/x64/src/regexp/regexp.go:603 +0x5d
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.resolveProperty({0x0, 0xc00034ab70, 0xc000e2ef40, 0x0, 0xc00034abb0, 0x0, 0x0, 0x0, 0xc00034abc0, 0x0, ...}, ...)
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:152 +0x59
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.newPackageFromPom({0x0, 0xc00034ab70, 0xc000e2ef40, 0x0, 0xc00034abb0, 0x0, 0x0, 0x0, 0xc00034abc0, 0x0, ...}, ...)
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:76 +0xc5
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/java.parserPomXML({0xc00050ba70?, 0xc00050b7c8?}, 0x415f50?, {{{{{...}, {...}}, {0x0, 0x0}, {0x105, {...}}}, {0xc0000237d0}}, ...})
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/java/parse_pom_xml.go:33 +0x351
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger/generic.(*Cataloger).Catalog(0xc000663650, {0x1b4d5d0, 0xc0001d26c0})
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/generic/cataloger.go:129 +0x74e
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.runCataloger({0x1b412d8, 0xc000663650}, {0x1b4d5d0?, 0xc0001d26c0})
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:65 +0x1fa
[INFO]     [creator]       github.com/anchore/syft/syft/pkg/cataloger.Catalog.func1()
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:146 +0xec
[INFO]     [creator]       created by github.com/anchore/syft/syft/pkg/cataloger.Catalog
[INFO]     [creator]       	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:141 +0x34a
[INFO]     [creator]       
[INFO]
```

**Comment by modulo11** [comment]
I think the [issue](https://github.com/anchore/syft/issues/2060) is fixed upstream with `v.088.0`. @dmikusa when would the next release of the buildpack happen?

**Comment by dmikusa** [comment]
I just published the release with 0.88.0. It should be available in a min or two. It'll get picked up in our Friday release cycle and pulled into the builders.
